frontend/src/components/visual-editor/v2/Diagrams.tsx

```diff
@@ -67,6 +69,13 @@ const Diagrams = () => {
   const [canvas, setCanvas] = useState<JSX.Element | undefined>();
   const [selectedBlockId, setSelectedBlockId] = useState<string | undefined>();
   const deleteDialogCtl = useDialog<string>(false);
+  const moveDialogCtl = useDialog<string[] | string>(false);
+  const { refetch: refetchBlocks } = useFind(
+    { entity: EntityType.BLOCK, format: Format.FULL },
+    {
+      hasCount: false,
+    },
+  );
```

Comment:
We already are refetching block in line 169, so you cann deconstruct the refetch function from there.
Otherwise I think it's better to invalidate/remove queries from the cache rather.

Reply:
Thanks :pray: 
Pushed the required changes.